src/test/java/com/es/phoneshop/model/dao/impl/ArrayListProductDaoTest.java

```diff
@@ -0,0 +1,75 @@
+package com.es.phoneshop.model.dao.impl;
+
+import com.es.phoneshop.model.dao.ProductDao;
+import com.es.phoneshop.model.entity.Product;
+import com.es.phoneshop.model.exception.PhoneShopException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class ArrayListProductDaoTest {
+    private static final ProductDao productDao = new ArrayListProductDao();
+    private static final Currency USD = Currency.getInstance("USD");
+
+    @BeforeClass
+    public static void setup() {
+        productDao.save(new Product("sgs", "Samsung Galaxy S", new BigDecimal(100), USD, 100, "manufacturer/Samsung/Samsung%20Galaxy%20S.jpg"));
```

Comment:
The approach for mocking product database in this one should be the same as for mocking ProductDao in servlet class. Speaking about protected-setter approach, ArrayListProductDao class could have a protected setter for phoneStock field which would be invoked by the test class (with mock object being passed). And the list of products returned by mock-PhoneStock should be also a List of mock-Product objects.